force-app/main/default/classes/FlowHandler.cls

```diff
@@ -0,0 +1,35 @@
+public with sharing class FlowHandler {
+
+    @invocableMethod (label='Check for Interview Overlap' description='Check for interview overlap with existing events')
+    public static List<Boolean> checkForInterviewOverlap(List<List<DateTime>> interviewStartEndTimes) {
+        List<Boolean> overLapFound = new List<Boolean>();
+        
+        for (List<DateTime> times : interviewStartEndTimes) {
+            List<Boolean> overlaps = new List<Boolean>();
+            DateTime startTime = times[0];
+            DateTime endTime = times[1];
+            Date interviewDate = startTime.Date();
+            // Find all events that are scheduled for the same date as the interview
+            List<Event> events = [SELECT Id, Subject, StartDateTime, EndDateTime FROM Event WHERE DAY_ONLY(StartDateTime) = :interviewDate];
```

Comment:
You might be able to optimize this query further by filtering for events where StartDateTime <= :endTime AND EndDateTime >= :startTime